packages/notification-center/src/store/notifications-provider.context.tsx

```diff
@@ -1,195 +1,129 @@
-    messageId: string,
-    actionButtonType: ButtonTypeEnum,
-    status: MessageActionStatusEnum,
-    payload?: Record<string, unknown>
-  ) {
-    await api.updateAction(messageId, actionButtonType, status, payload);
+import { NotificationsContext } from './notifications.context';
+import type { IStore } from '../shared/interfaces';
+import { useFetchNotifications, useUnseenCount } from '../hooks';
+import { useMarkNotificationsAs } from '../hooks/use-mark-notification-as.hook';
+
+const DEFAULT_STORES = [{ storeId: 'default_store' }];
+
+export function NotificationsProvider({
+  children,
+  stores = DEFAULT_STORES,
+}: {
+  children: React.ReactNode;
+  stores?: IStore[];
+}) {
+  const firstStore = stores[0];
+  const [storeQuery, setStoreQuery] = useState<IStoreQuery>(() => firstStore.query ?? {});
+  const [storeId, setStoreId] = useState(firstStore.storeId ?? 'default_store');
+  const setStore = useCallback(
+    (newStoreId: string) => {
+      const foundQuery = stores?.find((store) => store.storeId === newStoreId)?.query || {};
+      setStoreId(newStoreId);
+      setStoreQuery(foundQuery);
+    },
+    [stores, setStoreId, setStoreQuery]
+  );
+  const {
+    data: notificationsPages,
+    hasNextPage,
+    isLoading,
+    isFetching,
+    isFetchingNextPage,
+    fetchNextPage,
+    refetch,
+  } = useFetchNotifications({ query: storeQuery });
+  const { data: unseenCountData } = useUnseenCount();
+  const { markNotificationsAs } = useMarkNotificationsAs();
+
+  const markNotificationAsRead = useCallback(
+    (messageId: string) => markNotificationsAs({ messageId, seen: true, read: true }),
+    [markNotificationsAs]
+  );
 
-    for (const storeId in notifications) {
-      notifications[storeId] = notifications[storeId].map((message) => {
-        if (message._id === messageId) {
-          message.cta.action.status = MessageActionStatusEnum.DONE;
-        }
+  const markNotificationAsSeen = useCallback(
+    (messageId: string) => markNotificationsAs({ messageId, seen: true, read: false }),
+    [markNotificationsAs]
+  );
 
-        return message;
-      });
+  const markAllNotificationsAsRead = useCallback(() => {
+    if (!notificationsPages) {
+      return;
     }
 
-    setNotifications(Object.assign({}, notifications));
-  }
+    const messageIds = notificationsPages.pages.reduce<string[]>((acc, paginatedResponse) => {
+      const pageMessageIds = paginatedResponse.data.filter((message) => !message.read).map((message) => message._id);
 
-  async function refetch(storeId = 'default_store') {
-    setFetching(true);
+      return [...acc, ...pageMessageIds];
+    }, []);
 
-    if (refetchTimeout.get(storeId)) {
-      clearTimeout(refetchTimeout.get(storeId));
-      setRefetchTimeout(refetchTimeout.set(storeId, null));
+    if (messageIds.length > 0) {
+      markNotificationsAs({ messageId: messageIds, seen: true, read: true });
     }
+  }, [markNotificationsAs, notificationsPages]);
```

Comment:
Mark all notifications as read.
The logic is to find all the notifications that are not read from the ones that were fetched and then mark them as read and seen.